src/fonduer/parser/parser.py

```diff
@@ -262,6 +262,7 @@ def apply(  # type: ignore
             if self.visual:
                 # Use the provided pdf_path if present
                 self.pdf_path = pdf_path if pdf_path else self.pdf_path
+                self.vizlink.pdf_path = self.pdf_path
```

Comment:
Is this line is needed?
`pdf_path` is already set at https://github.com/HazyResearch/fonduer/blob/8edfacd4eabaa9fd49ac71bd63488322fa4b5fa2/src/fonduer/parser/parser.py#L250

Reply:
This is a good point. I think it's important since in the `apply` function we expose one `pdf_path` argument for users which will override the `vizlink.pdf_path` if it's not None.

Reply:
Ah, you are right.
Can you pass `pdf_path` as an argument to `self.vizlink.link` instead? It is hacky to directly change a class variable and and not thread-safe in this case.

Reply:
Agreed! Addressed.